src/main/java/frc/robot/subsystems/vision/VisionSubsystem.java

```diff
@@ -19,24 +22,21 @@ public class VisionSubsystem extends SubsystemBase {
   private Pose2d lastSeenPose = new Pose2d();
   private double headingDegrees = 0;
   private double headingRateDegreesPerSecond = 0;
-  private final Map<Integer, AtomicBoolean> runningThreads = new ConcurrentHashMap<>();
-  private final ExecutorService executorService =
-      Executors.newFixedThreadPool(
-          3); // Thread pool size of 3 for the number of limelights being threaded
+  private final Map<Integer, AtomicBoolean> limelightThreads = new ConcurrentHashMap<>();
+  private final ScheduledExecutorService scheduledExecutorService =
+      Executors.newScheduledThreadPool(3); // lol no clue if this'll work
```

Comment:
this is what I was talking about. dyk yet if this works?

Reply:
Copied from below - That wasn't changed for threading. I added that to schedule a new task if a limelight is disconnected. If u go to handleLimelightDisconnect u will see a big comment explaining what it should do (schedule something that should try to reconnect to the limelight after 5 seconds). I'm thinking I could also use a timer to accomplish that same thing tho idk, this seems fancier 🤷‍♂️.

Reply:
No clue if it will work